### PR TITLE
Exceptions in Java exercises

### PR DESCRIPTION
**Pull request by dskrumov**
These are the 3 tasks from exceptions in java. 

Please make a review @mgenov .

### CODE REVIEW

**Comment by mgenov** [comment]
Please squash your comments.

File: .idea/misc.xml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mgenov) [inline .idea/misc.xml]: IDEA project files should be added to .gitignore. 

https://github.com/github/gitignore/blob/master/Global/JetBrains.gitignore

File: .idea/vcs.xml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="$PROJECT_DIR$" vcs="Git" />
+  </component>
+</project>
```
Comment (mgenov) [inline .idea/vcs.xml]: Class files should be added to .gitignore as everyone could compile them on it's own PC.

File: src/com/company/arrayoperation/Array.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            if (array[i] == 0) {
+                throw new arrayRangeException("You can not remove a element from an empty Array");
+            }
+        }
+    }
+
+
+}
+
+
```
Comment (mgenov) [inline src/com/company/arrayoperation/Array.java]: these empty lines need to be removed

File: src/com/company/arrayoperation/Array.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    /**
+     * Remove the last element of the array.
+     * @throws arrayRangeException is the custom exception
+     */
+    public void removeElements() throws arrayRangeException {
+
+        for (int i = 0; i < array.length - 1; i++) {
+            System.out.print(" " + array[i]);
+            if (array[i] == 0) {
+                throw new arrayRangeException("You can not remove a element from an empty Array");
```
Comment (mgenov) [inline src/com/company/arrayoperation/Array.java]: Class names in java should start with upper case character.

File: src/com/company/arrayoperation/Array.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
+package com.company.arrayoperation;
+
+
+public class Array {
+    /**
+     * Fields
+     */
+    private int[] array;
+    private int val;
```
Comment (mgenov) [inline src/com/company/arrayoperation/Array.java]: val = value from my understanding, but it is used for indexing. Name need to be improved.

File: src/com/company/arrayoperation/Array.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    public Array(int size) {
+        array = new int[size];
+    }
+
+    /**
+     * Add method to add values in the array
+     * @param Array is the array
+     * @throws arrayRangeException is the custom exception
+     */
+    public void add(int Array) throws arrayRangeException {
```
Comment (mgenov) [inline src/com/company/arrayoperation/Array.java]: variables and name of the references to be in camelCase starting with lower case

array, userName, depositAccount and etc. 

https://github.com/clouway/teamspace/wiki/Java%20Code%20Convention

File: src/com/company/arrayoperation/Array.java [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+        }
+    }
+
+    /**
+     * Remove the last element of the array.
+     * @throws arrayRangeException is the custom exception
+     */
+    public void removeElements() throws arrayRangeException {
+
+        for (int i = 0; i < array.length - 1; i++) {
```
Comment (mgenov) [inline src/com/company/arrayoperation/Array.java]: Doesn't look like a good solution as it iterates over all elements. Think for better one which is more optimized.

**Comment by dskrumov** [comment]
Already made the changes in the array and the code is reformated and renamed. I already ignored the files and the folders but they still appear here. Dont know how to deal with it. 

Please @mgenov make a review. Thank you.

**Comment by mgenov** [comment]
https://www.google.bg/#q=.gitignore+committed+file&*

File: src/com/company/arrayoperation/Array.java [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+        for (int i = 0; i < array.length; i++) {
+            System.out.print(" " + array[i]);
+        }
+    }
+    /**
+     * Remove the last element of the array.
+     *
+     * @throws ArrayRangeException is the custom exception
+     */
+    public void removeElements() throws ArrayRangeException {
```
Comment (mgenov) [inline src/com/company/arrayoperation/Array.java]: removeElement() or removeLast()

File: src/com/company/arrayoperation/Array.java [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+        int[] array2 = Arrays.copyOf(array, array.length - 1);
```
Comment (mgenov) [inline src/com/company/arrayoperation/Array.java]: This is not going to work as expected as the last element is not removed from the array that is wrapped.

**Pull request closed without merging**